labs/b10.md

```diff
@@ -0,0 +1,157 @@
+the differences between blocks in a file, and only transferring the difference (the 'delta'). By default, it assumes that a file is unchanged if its last modified time is not newer.
+
+The basic syntax of `rsync` is `rsync [options] [[user]@host:]source [[user]@host:]dest` - you can either transfer files
+in the local filesystem, between the local system and a network host, or between network hosts. 
+
+`rsync` has dozens of options, but the most common are the following:
+  * `-r` `--recursive`: recursively transfer all files and subdirectories in the source
+  * `-v` `--verbose`: verbosely print out file names and transfer speeds
+  * `-P` `--partial --progress`: keep partially transferred files if the transfer is interrupted, show progress of file transfers
+  * `-a` `--archive`: combination option, implies `-r`, and preservation of most file metadata properties
+  * `-u` `--append`: append data to the end of files on the destination that are smaller than the corresponding
+                     file on the source (useful for log files, dangerous if used on files that do not only grow by appending data)
+  * `-z` `--compress`: compress files before transfer (Only useful for `rsync`ing over the network)
+                     
+Suppose you keep all your homework in a directory "school", in your home directory:
+```
+user@machine [~/school] $ ls
+cs198-8
+cs61a
+cs262
+some_r1a
+ds8
+```
+and you want to back everything to your student VM. A command like:
+
+    $ rsync -rvPaz ~/school user@user.decal.xcf.sh:school-backup/
+    
+would result in a copy of your `school` directory being made inside the `school-backup` directory on your student VM.
+
+Play around with `rsync` yourself. We've provided some test files [here](https://decal.ocf.berkeley.edu/static/b10/b10.tar.gz). (hint: use `wget`).
+After extracting the archive, (`tar xzvf b10.tar.gz`), you'll see two directories with over 100MB of files in them.
+
+* What happens if you try `rsync`ing `dir1` to `dir2`? (hint: it should finish almost instantaneously). Why might this be the case?
+
+* What `rsync` commands would you use to keep both directories in sync, from updating the source to dest and dest to source?
+
+* What happens if you truncate one of the files in the dest directory (e.g. `dir2`, by `truncate --size=5M file2.txt`) and then try `rsync` from source to dest again?
+
+## Conclusion
+
+Now that you know how to set up a Gitolite server and add users and repositories to it, you can now use your VM (before we delete it)
+as a git remote to back up your programming projects, fanfic collections, etc., free from the prying eyes of our Github overlords.
+
+## Checkoff
+
+The checkoff form is [here](https://goo.gl/forms/8QMYcjnHHQQT426A3). Please double-check that you've completed the Gitolite section successfully, as we will be verifying that when checking you off.
```

Comment:
Maybe say that we will verify it by trying to connect over SSH so that they can verify it themselves by adding a separate key or something?